### PR TITLE
HIA 661 persons search 500 with pnc bug

### PR DESCRIPTION
**Pull request by popey2700**
## Context

When searching persons with PNC if the probation offender search doesn’t return a person the request will crash into a 500 rather than a 404. As a result of Prisoner search not taking a PNC we decided to change functionality so if someone searches with PNC they will only get ProbationOffender results back

## Changes proposed in this PR
- Changed PNC search flow

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/prisoneroffendersearch/PrisonerOffenderSearchGatewayTest.kt [hunk 1]
```diff
@@ -108,42 +106,13 @@ class PrisonerOffenderSearchGatewayTest(
...
 
-      val response = prisonerOffenderSearchGateway.getPersons("Obi-Wan", null, null, null)
+      val response = prisonerOffenderSearchGateway.getPersons("Obi-Wan", null, null)
 
       response.data.count().shouldBe(1)
       response.data.first().firstName.shouldBe("Obi-Wan")
       response.data.first().lastName.shouldBe("Kenobi")
     }
 
-    it("returns person(s) when searching on hmppsId only") {
```
Comment (AlexHerbertMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/prisoneroffendersearch/PrisonerOffenderSearchGatewayTest.kt:118]: Was this removed on purpose?

**Review by AlexHerbertMoJ — COMMENTED** [review]
Not sure I'm the best person to review; I'd appreciate it if you had time to talk me through the changes you made and why they work

Additionally, I don't see a test case for the situation presented in the bug. Again, I'm still acclimatising to this team's ways of working, but would it not be good to make a test for the specific case to ensure it doesn't crop up again in the future?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/prisoneroffendersearch/PrisonerOffenderSearchGatewayTest.kt (hunk 1 above)
Comment (CslBing) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/prisoneroffendersearch/PrisonerOffenderSearchGatewayTest.kt:118]: Yes, we have excluded pnc search via prisoner, as it is not a search term used by prisoner

**Review by AlexHerbertMoJ — APPROVED** [review]
Looks good to me :)

**Pull request merged**